### PR TITLE
Consumables rework

### PR DESCRIPTION
**Pull request by Draykoss**
Reworked UseSlot and expanded code functionality for all consumable items

### CODE REVIEW

File: Protocols.External/[h] HotbarHandler/[u] UseSlot.cs [hunk 1]
```diff
@@ -167,7 +147,8 @@ private void DropItem(object data)
                     {
                         PrefabName = prefabName,
                         Component = component,
-                        ObjectId = objectId
+                        ObjectId = objectId,
+                        Damage = dropData.UsedItem.ItemEffects.FirstOrDefault().Value
```
Comment (FeroxFoxxo) [inline Protocols.External/[h] HotbarHandler/[u] UseSlot.cs:151]: Create method to calculate damage using effects (i.e. look for PhysicalDamage to start. Could also calculate if other effects matter (i.e. poison etc) in the future

File: Protocols.External/[h] HotbarHandler/[u] UseSlot.cs [hunk 2]
```diff
@@ -238,20 +209,17 @@ private void HandleMeleeWeapon(ItemDescription usedItem, Vector3Model position,
...
-        var meleeWeapon = ItemCatalog.GetItemFromId(usedItem.ItemId);
-        var weaponPrefabName = meleeWeapon.PrefabName;
-
-        var meleeSyncEvent = new Melee_SyncEvent(Player.GameObjectId.ToString(), Player.Room.Time,
-            position.X, position.Y, position.Z, 0, 0, 100, 0, weaponPrefabName);
-
-        Player.Room.SendSyncEvent(meleeSyncEvent);
+        var weaponDamage = 20; //Temp value if weapon damage value is null.
+        if (usedItem.ItemEffects.Count > 0)
+            weaponDamage = usedItem.ItemEffects.FirstOrDefault().Value; //Replace with xml system?
```
Comment (FeroxFoxxo) [inline Protocols.External/[h] HotbarHandler/[u] UseSlot.cs:222]: Same as above, don't use first or default as the first might be a different item effect than physical damage

File: Protocols.External/[h] HotbarHandler/[u] UseSlot.cs [hunk 3]
```diff
@@ -281,10 +249,11 @@ private void HandleMeleeWeapon(ItemDescription usedItem, Vector3Model position,
                     else if (component is BreakableEventControllerComp breakableObjEntity)
                         breakableObjEntity.Destroy(Player);
                     else if (component is InterObjStatusComp enemyEntity)
-                        enemyEntity.SendDamageEvent(Player);
+                        enemyEntity.SendDamageEvent(Player, weaponDamage);
         }
     }
 
+
```
Comment (FeroxFoxxo) [inline Protocols.External/[h] HotbarHandler/[u] UseSlot.cs:256]: Random new line character

File: Protocols.External/[i] InventoryHandler/[u] UseItem.cs [hunk 4]
```diff
@@ -83,25 +79,118 @@ public override void Run(string[] message)
...
+        position.Z = 0;
+
+        var dropItemData = new DroppedItemData()
+        {
+            DropDirection = dropDirection,
+            Position = position,
+            UsedItem = usedItem
+        };
+
+        TimerThread.DelayCall(DropItem, dropItemData, TimeSpan.FromMilliseconds(1000), TimeSpan.Zero, 1);
```
Comment (FeroxFoxxo) [inline Protocols.External/[i] InventoryHandler/[u] UseItem.cs:122]: Is this delay consistent across all bombs? Maybe move to a config file if so

File: Protocols.External/[i] InventoryHandler/[u] UseItem.cs [hunk 5]
```diff
@@ -83,25 +79,118 @@ public override void Run(string[] message)
...
+                {
+                    var bombData = new BombData()
+                    {
+                        PrefabName = prefabName,
+                        Component = component,
+                        ObjectId = objectId,
+                        Damage = dropData.UsedItem.ItemEffects.FirstOrDefault().Value
+                    };
+
+                    TimerThread.DelayCall(ExplodeBomb, bombData, TimeSpan.FromMilliseconds(2650), TimeSpan.Zero, 1);
```
Comment (FeroxFoxxo) [inline Protocols.External/[i] InventoryHandler/[u] UseItem.cs:160]: Move to a config file if consistent across all bombs

File: Server.Reawakened/Players/Extensions/PlayerExtensions.cs [hunk 6]
```diff
@@ -110,6 +114,10 @@ public static void RemoveTrade(this Player player)
     public static void AddBananas(this Player player, int collectedBananas)
     {
         var charData = player.Character.Data;
+
+        if (player.TempData.BananaBoostsElixir)
+            collectedBananas = Convert.ToInt32(collectedBananas * 0.1);
```
Comment (FeroxFoxxo) [inline Server.Reawakened/Players/Extensions/PlayerExtensions.cs:119]: Move to config file

Comment (Draykoss) [reply Server.Reawakened/Players/Extensions/PlayerExtensions.cs:119]: will do this later. out of time for now

**Review by FeroxFoxxo — APPROVED** [review]
LGTM

**Pull request merged**